cheroot/__main__.py

```diff
@@ -0,0 +1,103 @@
+import argparse
+from importlib import import_module
+import os
+import sys
+
+import six
+
+from .wsgi import Server
+
+
+def wsgi_app_path(string):
+    if ':' in string:
+        mod_path, app_path = string.split(':', 1)
+    else:
+        mod_path, app_path = string, 'application'
+
+    return mod_path, app_path
+
+
+def wsgi_bind_addr(string):
+    if ':' in string:
```

Comment:
Mind adding ability to bypassing UNIX sockets via argument?

Here's how curl, httpie and requests do it:
```
curl --unix-socket /tmp/cp.sock http://localhost/
http http+unix://%2Fvar%2Frun%2Fdocker.sock/info
r = requests.get('http+unix://%2Fvar%2Frun%2Fdocker.sock/info')
```

Also, software tends to prefix abstract sockets with `@`:
https://unix.stackexchange.com/questions/206386/what-does-the-symbol-denote-in-the-beginning-of-a-unix-domain-socket-path-in-l?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

So I'd like it to accept unix socket starting with an `@` (or extra argument if you think it's better)

Reply:
I had a long think about this... the two supported inputs for bind_addr are either the tuple (host_or_ip, port), or a string as a path for a UNIX socket. As the string equivalent for the former ("host_or_ip:port") is pretty easy to match with a regex, it's possible to get away with treating any non-matches as a UNIX socket path. Main difference would be that passing just an IP address would no longer give you port 8000 for free, but chances are if you want to change the IP address you'll be wanting to change the port too.

Alternatively we could put in a case to deal with just a bare IP address so it doesn't make a socket, or require UNIX socket paths begin with '@' and throw an exception as the overall fallback. Also if the plan is to make a new struct type for bind_addr, any string deserializing logic should go in there.